File: go-ms/domain/usecase/db_use_case.go
package usecase

import (
	model "maocq/go-ms/domain/model/account"
	account "maocq/go-ms/domain/model/account/gateways"
)

type DBUseCase struct {
	AccountRepository account.AccountRepository
}

func (c *DBUseCase) Query() *model.Account {
	return c.AccountRepository.FindById(4000)
}

File: java-imperative-ms/domain/usecase/src/main/java/com/maocq/usecase/db/DBUseCase.java
package com.maocq.usecase.db;

import com.maocq.model.account.Account;
import com.maocq.model.account.gateways.AccountRepository;
import lombok.RequiredArgsConstructor;

import java.util.Optional;

@RequiredArgsConstructor
public class DBUseCase {

  private final AccountRepository accountRepository;
  public Optional<Account> query() {
    return accountRepository.findById(4000);
  }
}
